astro-docs/src/plugins/utils/devkit-generation.ts

```diff
@@ -0,0 +1,161 @@
+  // generate ngcli docs in same dir
+  const ngcliEntryPoint = join(
+    workspaceRoot,
+    'dist',
+    'packages',
+    'devkit',
+    'ngcli-adapter.d.ts'
+  );
+  if (existsSync(ngcliEntryPoint)) {
+    await runTypeDoc(
+      {
+        ...defaultTypedocOptions,
+        entryPoints: [ngcliEntryPoint],
+        tsconfig: join(buildDir, 'tsconfig.lib.json'),
+        out: join(outDir, 'ngcli_adapter'),
+        publicPath: '/docs/reference/devkit/ngcli_adapter/',
+      },
+      logger
+    );
+  }
+
+  logger.info(`Loading devkit docs from ${outDir}`);
+
+  const markdownFiles = await walkDirectory(outDir);
+
+  for (const filePath of markdownFiles) {
+    // Get the relative path from the output directory
+    const relativePath = filePath.replace(outDir + '/', '');
+    // Get the title from the filename (last part without .md)
+    const pathParts = relativePath.split('/');
+    const fileName = pathParts[pathParts.length - 1];
+
+    let slug = '';
+    let title = '';
+    let category = '';
+
+    // Handle README.md files as overview routes
+    if (fileName === 'README.md') {
+      if (pathParts.length === 1) {
+        // Root README becomes 'overview' 'devkit' index route
+        slug = '';
+        title = '@nx/devkit Overview';
+        category = 'overview';
+      } else if (pathParts.length === 2 && pathParts[0] === 'ngcli_adapter') {
+        // ngcli_adapter/README becomes 'devkit/ngcli_adapter' index route
+        slug = 'ngcli_adapter';
+        title = 'ngcli_adapter Overview';
+        category = 'overview';
+      }
+    } else {
+      // Regular markdown files are in the same location as file location
+      // NOTE: we might want to change this to flatten out the "types" i.e. interface/function etc
+      // that will require changing the URL generation of typedoc as well
+      slug = relativePath.replace(/\.md$/, '');
+      title = fileName.replace(/\.md$/, '');
+      category = pathParts.length > 1 ? pathParts[0] : 'overview';
+    }
+
+    // Read the markdown content
+    let content = await readFile(filePath, 'utf-8');
+
+    // // Remove .md extensions from all links in the content
+    if (content) {
+      // Remove .md from markdown links: [text](path.md) -> [text](path)
+      content = content
+        .replace(/(\[.*?\]\([^)]*?)\.md(\)|#)/gi, '$1$2')
+        // Remove .md from any remaining URLs that end with .md
+        .replace(/\.md(?=[#)\s]|$)/gim, '');
+    }
+
+    const rendered = content ? await renderMarkdown(content) : undefined;
+
+    const mappedCategory = directoryToCategoryMap[category] || category;
+
+    const documentRecord: CollectionEntry<'nx-reference-packages'> = {
+      id: `devkit_${slug.replace(/\//g, '-')}`,
+      body: content,
+      // @ts-expect-error astro auto gen types don't align
+      rendered,
+      collection: 'nx-reference-packages',
+      data: {
+        title: title,
+        packageType: 'devkit',
+        docType: slug,
```

Comment:
Should we just add `slug` to the schema here? Looks like it has both `docType` and `category` in the schema, but I'm unsure what the difference is, as both seem to be for `generators`, `overview`, etc. But in this context the `docType` is being used as a slug.